client/src/main/java/org/client/App.java

```diff
@@ -19,51 +21,88 @@
     private Socket sock;
     private PrintWriter out;
     private BufferedReader in;
-    private final int EOT = 0x04;
-    
+
+    /**
+     * Main entry point of the client application.
+     * 
+     * @param args
+     */
     public static void main( String[] args )
     {
         App client = new App();
         client.start();
-        
+
         // temporary, we will run this in a loop
         Console console = System.console();
         console.readLine("Any key to exit.");
     }
-    
-    // start the client logic
+
+    /**
+     * Start the client logic.
+     */
     void start() {
-        
+
         try {
             this.sock = new Socket(serverAddress, serverPort);
-            out = new PrintWriter(this.sock.getOutputStream(), true);
-            in = new BufferedReader(
+            this.out = new PrintWriter(this.sock.getOutputStream(), true);
+            this.in = new BufferedReader(
                     new InputStreamReader(sock.getInputStream()));
         } catch (UnknownHostException e) {
-            // TODO Auto-generated catch block
+            System.out.println("Host \"" + serverAddress + "\" is unknown.");
             e.printStackTrace();
+            return;
         } catch (IOException e) {
-            // TODO Auto-generated catch block
+            System.out.println("Failed to connect. Is server running at \"" + serverAddress + ":" + serverPort + "\"?");
             e.printStackTrace();
+            return;
         }
-        
+
         // Now we know the server is online.
-    
-        // get their desired username
+
+        // get their desired username. want to loop until they find an acceptable
+        // username.
+        boolean loginSuccessful = false;
         Console console = System.console();
-        String username = console.readLine("Enter your username: ");
-        String loginMsg = createLoginMessageForUser(username);
-        sendMessage(loginMsg);
-    }
-    
-    // Already know the address of the server, send the login message
-    private void sendMessage(String msg) {
-        msg = msg + Character.toString((char) EOT);
-        out.print(msg);
-        out.flush();
+        while(!loginSuccessful) {
+            String username = console.readLine("Enter your username: ");
+            String loginMsg = createLoginMessageForUser(username);
+            Utils.sendMessage(this.out, loginMsg);
+            if(statusOk()) {
+                System.out.println("Successful login as \"" + username + "\"");
+                // we shall stop trying
+                loginSuccessful = true;
+            }
+            else {
+                System.out.println("Failed to login as \"" + username + "\"");	
+            }
+        }
```

Comment:
at this point we should enter a loop of accepting user input to either interpret as a local command (to select remote contacts, change chat, log off) or get as a chat message and send it off.